Kudu.Core/Deployment/Generator/FunctionAppBuilder.cs

```diff
@@ -2,18 +2,38 @@
 
 namespace Kudu.Core.Deployment.Generator
 {
-    class FunctionAppBuilder : BaseBasicBuilder
+    class FunctionAppBuilder : GeneratorSiteBuilder
     {
-        public FunctionAppBuilder(IEnvironment environment, IDeploymentSettingsManager settings, IBuildPropertyProvider propertyProvider, string repositoryPath, string projectPath) 
-            :base(environment, settings, propertyProvider, repositoryPath, projectPath, "--functionApp")
+        private readonly string _projectFilePath;
+        private readonly string _solutionPath;
+
+        public FunctionAppBuilder(IEnvironment environment, IDeploymentSettingsManager settings, IBuildPropertyProvider propertyProvider, string repositoryPath, string projectFilePath, string solutionPath = null)
+            : base(environment, settings, propertyProvider, repositoryPath)
+        {
+            _projectFilePath = projectFilePath;
+            _solutionPath = solutionPath;
+        }
+
+        protected override string ScriptGeneratorCommandArguments
         {
+            get
+            {
+                if (string.IsNullOrEmpty(_solutionPath))
+                {
+                    return $"--functionApp \"{_projectFilePath}\"";
```

Comment:
So it's valid not to have a solution in the csproj case? I thought you said it wasn't.